portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,25 +77,32 @@ private int getNumComments(HttpServletRequest request) {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String commentText = request.getParameter("comment-input");
-    Entity commentEntity = createCommentEntity(commentText);
-    datastore.put(commentEntity);
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String commentText = request.getParameter("comment-input");
 
-    response.sendRedirect("/index.html");
+      Entity commentEntity = createCommentEntity(commentText, userEmail);
+      datastore.put(commentEntity);
+
+      response.sendRedirect("/index.html");
+    }
```

Comment:
Should an error code be returned if the user isn't logged in?

Reply:
Done.